satpy/multiscene.py

```diff
@@ -345,7 +345,8 @@ def load_data(q):
 
         for scene in scenes_iter:
             delayed = scene.save_datasets(compute=False, **kwargs)
-            if isinstance(delayed, (list, tuple)) and len(delayed) == 2:
+            sources, targets, _ = split_results(delayed)
```

Comment:
Shouldn't the `_` be overwriting `delayed`? This way, in the future, if the MultiScene can handle source/target combinations and delayed objects the code won't have to be changed. Right?

Reply:
Good point.

Reply:
Targets should never be returned without a source. They are source + target pairs. The simple image writer should return delayed objects. The geotiff writer returns source (dask array) + target (output file-like object). If you have another example, then I still propose that `_` be replaced with `delayed`.

Reply:
Ok I see now I got fooled by the test which returned targets without sources from `split_results` for the simple image writer because the mock did not pass as a delayed object. I changed the code to overwrite delayed now and also fixed the test. Additionally I added a test for writers with source/target combinations to fail.